### PR TITLE
Added Rich text editor

### PR DESCRIPTION
**Pull request by zoef-yt**
[PREVIEW LINK](https://deploy-preview-4--jot-it-down-zoef.netlify.app/)
In this PR, I added
Rich text editor.
Left some commented code for future references

image

### CODE REVIEW

File: src/Components/NotesPage/MyEditor.css [hunk 1]
```diff
@@ -0,0 +1,70 @@
+.editor-class {
+	background-color: var(--colour-black-third);
+	color: var(--colour-text-white);
+	display: block;
+	border-radius: 0 0 0.5rem 0.5rem;
```
Comment (iRohitGaur) [inline src/Components/NotesPage/MyEditor.css]: suggestion: 0.5rem can be written in px. Value too small to keep in rem.

File: src/Components/NotesPage/MyEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+				wrapperClassName='wrapper-class'
+				editorClassName='editor-class'
+				toolbarClassName='toolbar-class'
+				onEditorStateChange={onEditorStateChange}
+				hashtag={{
+					separator: ' ',
+					trigger: '#',
+				}}
+				toolbar={{
+					// options: ['inline', 'list', 'textAlign', 'history', 'link'],
```
Comment (iRohitGaur) [inline src/Components/NotesPage/MyEditor.jsx:20]: suggestion: remove commented code

**Review by iRohitGaur — APPROVED** [review]
Nice work!

UI looks great.

Good to merge 👍🏻

File: src/Components/NotesPage/MyEditor.css [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	align-items: center;
+	padding: 1rem 0.2rem;
+	margin: 0;
+	box-shadow: none;
+}
+
+.my-editor .rdw-option-wrapper {
+	background-color: transparent;
+	border-color: var(--colour-primary);
+	border-radius: 0.5rem;
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/MyEditor.css]: Suggestion: Border radius can be given in px

File: src/Components/NotesPage/NotesPage.jsx [hunk 4]
```diff
@@ -1,38 +1,85 @@
...
+	const [editorState, setEditorState] = useState(EditorState.createEmpty());
 	const submitNote = () => {
 		postNotes({
 			title: textField.title,
 			body: textField.body,
 			tags: [],
 			createdAt: formatDate(),
 		});
+		setEditorState(EditorState.createEmpty());
+		setTextField({ title: '', body: '' });
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx]: Can declare this as an initial variable.

File: src/Components/NotesPage/NotesPage.jsx [hunk 5]
```diff
@@ -1,38 +1,85 @@
...
+			</div>
+
 			{notes.length > 0 ? (
 				<div>
 					{notes.map((note) => {
 						return (
 							<div className='card ' key={note._id}>
 								<h3>{note.title}</h3>
-								<p>{note.body}</p>
+								<div style={{ width: '100%' }} className=' rdw-editor-main' dangerouslySetInnerHTML={{ __html: note.body }}></div>
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx]: Please avoid inline styling.

File: src/Components/NotesPage/NotesPage.jsx [hunk 6]
```diff
@@ -1,38 +1,85 @@
...
+				<div className='flex-row justify-content-center'>
+					<button className='btn btn-primary' onClick={submitNote}>
+						Add
+					</button>
+				</div>
+			</div>
+
 			{notes.length > 0 ? (
 				<div>
 					{notes.map((note) => {
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx:80]: Directly destructure.

File: src/Components/NotesPage/NotesPage.jsx [hunk 7]
```diff
@@ -1,38 +1,85 @@
...
 		postNotes({
 			title: textField.title,
 			body: textField.body,
 			tags: [],
 			createdAt: formatDate(),
 		});
+		setEditorState(EditorState.createEmpty());
+		setTextField({ title: '', body: '' });
 	};
 	const onChangerHandler = (e) => {
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/NotesPage.jsx]: Please rename this to titleChangeHandler.

File: src/Components/NotesPage/MyEditor.css [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+.my-editor > * {
+	margin: 1rem 0;
+}
+
+.rdw-editor-toolbar > div {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	padding: 1rem 0.2rem;
```
Comment (muskanbagrecha) [inline src/Components/NotesPage/MyEditor.css:50]: 0.2 rem can be given in px.

**Review by muskanbagrecha — APPROVED** [review]
Really great job with the rich text editor. Love how the end result has come out. Absolutely love the image embedding. 
Suggestion: 
image
Please change the text color as the current one has poor contrast.

**Review by Shubhamkr2610 — COMMENTED** [review]
after making correction of suggestion by other folks , you are good to go

File: src/Components/NotesPage/MyEditor.css [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+	margin: 0;
```
Comment (varun13169) [inline src/Components/NotesPage/MyEditor.css:51]: Is there any reason to mention margin as 0 explicitly?

File: src/Components/NotesPage/NotesPage.jsx (hunk 5 above)
Comment (varun13169) [inline src/Components/NotesPage/NotesPage.jsx]: Create a helper class to avoid inline styling.

**Review by varun13169 — APPROVED** [review]
The code looks clean and good.
Made a few comments for better maintainability.

File: src/Components/NotesPage/MyEditor.css (hunk 1 above)
Comment (zoef-yt) [reply src/Components/NotesPage/MyEditor.css]: done

File: src/Components/NotesPage/MyEditor.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Components/NotesPage/MyEditor.jsx:20]: This is for future reference.

File: src/Components/NotesPage/MyEditor.css (hunk 3 above)
Comment (zoef-yt) [reply src/Components/NotesPage/MyEditor.css]: done

File: src/Components/NotesPage/NotesPage.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Components/NotesPage/NotesPage.jsx]: done

File: src/Components/NotesPage/MyEditor.css (hunk 9 above)
Comment (zoef-yt) [reply src/Components/NotesPage/MyEditor.css:51]: The styling is coming from the npm package so overriding it over here.

File: src/Components/NotesPage/NotesPage.jsx (hunk 7 above)
Comment (zoef-yt) [reply src/Components/NotesPage/NotesPage.jsx]: done

File: src/Components/NotesPage/NotesPage.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Components/NotesPage/NotesPage.jsx:80]: done

File: src/Components/NotesPage/MyEditor.css (hunk 8 above)
Comment (zoef-yt) [reply src/Components/NotesPage/MyEditor.css:50]: done

**Pull request merged**